src/variables/components/VariableDropdownReactSelect.tsx

```diff
@@ -0,0 +1,369 @@
+    }
+
+    this.state = defaultState
+  }
+
+  // set the 'shownValues' after loading, and
+  // resets the menuOpen variable
+  componentDidUpdate(prevProps, prevState) {
+    const {values, selectedValue, status} = this.props
+    const {values: prevVals, status: prevStatus} = prevProps
+    const {actualVal, loaded, selectHappened, menuOpen} = this.state
+    const {
+      actualVal: prevActualVal,
+      selectHappened: prevSelectHappened,
+      menuOpen: prevMenuOpen,
+    } = prevState
+
+    let newState = {}
+
+    const justLoaded = () =>
+      status === RemoteDataState.Done && prevStatus !== RemoteDataState.Done
+
+    const justSelected = () =>
+      selectHappened &&
+      !prevSelectHappened &&
+      actualVal &&
+      actualVal !== prevActualVal
+
+    // this is for updating the values:
+    // (only want this to run *once* when the values get loaded)
+    if (justLoaded()) {
+      newState = this.getInitialValuesAfterLoading()
+    } else if (
+      !loaded &&
+      status === RemoteDataState.Done &&
+      prevVals.length !== values.length
+    ) {
+      newState = {
+        shownValues: values,
+        typedValue: selectedValue,
+        loaded: true,
+      }
+    }
+
+    /**
+     * unset the menuOpen; it should be set to closed (or open) only once; then undone
+     * this is needed because: (from clockface Dropdown.tsx documentation):
+     * if the string is set to 'open', and then the user closes it, and the code sets it to open again,
+     * unless the code sets it to something else in between (like null or 'close'),
+     * then nothing will happen- the menu will not open)
+     */
+
+    if (menuOpen !== prevMenuOpen && menuOpen !== null) {
+      newState['menuOpen'] = null
+    }
+
+    // need to have this, as the 'onClickAwayHere' gets triggered *before*
+    // the selected value is set to the actualValue (it keeps re-using the original
+    // property)
+
+    // for updating the selected value:
+    if (justSelected()) {
+      newState = {...newState, typedValue: actualVal, selectHappened: false}
+    }
+
+    this.setState(newState)
+  }
+
+  getInitialValuesAfterLoading() {
+    const {values} = this.props
+    // it reloaded; maybe because of a dependent var
+    // want to re-init
+
+    return {
+      shownValues: values,
+      loaded: true,
+    }
+  }
+
+  render() {
+    const {selectedValue, name, status} = this.props
+    const {shownValues} = this.state
+
+    const isDisabled = !shownValues || shownValues.length === 0
+
+    let realVals = []
+
+    //  make them all look like: { value: 'vanilla', label: 'Vanilla' }
+    if (shownValues && shownValues.length) {
+      realVals = shownValues.map(val => ({value: val, label: val}))
+    }
+    const placeHolderText = this.getPlaceHolderText('Select a Value')
+
+    const foreground = 'white'
+    const inputBackground = 'black'
+    const background = '#2c2d35'
+    const focusColor = '#46454e'
+    const selectColor = '#ce58eb'
```

Comment:
where do these hex values come from? Clockface vends a set of color variables in the `InfluxColors` enum. `ce58eb` is represented by `Comet` and `#ffffff` is represented by `White`, but I didn't find mention of `#2c2d35` or `#46454e` or `#5d5f6f` or `#828497`

Reply:
i don't remember...i was going through and comparing colors and making the component look the same (as much as possible) as the existing component

Reply:
used influxcolors where I could; the rest I got from screencaps (matching colors)